internal/pkg/skuba/kubernetes/node_version_test.go

```diff
@@ -18,394 +18,1027 @@
 )
 
+const (
+	master0           = "master-0"
+	master1           = "master-1"
+	worker0           = "worker-0"
+	worker1           = "worker-1"
+	apiserver         = "kube-apiserver"
+	controllerManager = "kube-controller-manager"
+	scheduler         = "kube-scheduler"
+	etcd              = "etcd"
+)
+
+func TestAvailablePlatformVersions(t *testing.T) {
+	versions := StaticVersionInquirer{}.AvailablePlatformVersions()
+	for _, v := range versions {
+		t.Run(fmt.Sprintf("version(%v) should parse semantic", v), func(t *testing.T) {
+			if _, err := version.ParseSemantic(v.String()); err != nil {
+				t.Errorf("error not expected: (%v)", err)
+			}
+		})
+	}
+}
+
+func TestNodeVersionInfoForClusterVersion(t *testing.T) {
+	tests := []struct {
+		node corev1.Node
+	}{
+		{
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: master0,
+					Labels: map[string]string{
+						kubeadmconstants.LabelNodeRoleMaster: "",
+					},
+				},
+			},
+		},
+		{
+			node: corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: worker0,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+
+		var ver string
+		isMaster := IsControlPlane(&tt.node)
+		clusterVersions := StaticVersionInquirer{}.AvailablePlatformVersions()
+		for _, cv := range clusterVersions {
+			tName := fmt.Sprintf("%v node version info when cluster version is %v", tt.node.Name, cv)
+			t.Run(tName, func(t *testing.T) {
+				vInfo := StaticVersionInquirer{}.NodeVersionInfoForClusterVersion(&tt.node, cv)
+				if vInfo.Nodename == "" {
+					t.Error("node name expected, but none returned")
+				}
+				ver = vInfo.ContainerRuntimeVersion.String()
+				if _, err := version.ParseSemantic(ver); err != nil {
+					t.Errorf("container runtime version(%v) should parse semantic", ver)
+				}
+				ver = vInfo.KubeletVersion.String()
+				if _, err := version.ParseSemantic(ver); err != nil {
+					t.Errorf("kubelet version(%v) should parse semantic", ver)
+				}
+
+				if isMaster {
+					ver = vInfo.APIServerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("api server version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.ControllerManagerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("controller manager version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.SchedulerVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("scheduler version(%v) should parse semantic", ver)
+					}
+					ver = vInfo.EtcdVersion.String()
+					if _, err := version.ParseSemantic(ver); err != nil {
+						t.Errorf("etcd version(%v) should parse semantic", ver)
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestString(t *testing.T) {
+	cluster := "1.1.1"
+	tests := []struct {
+		name             string
+		isMaster         bool
+		apiServerVersion version.Version
+		kubeletVersion   version.Version
```

Comment:
Prefer to `*version.Version` to avoids dereference cause the `NodeVersionInfo struct` all are `*version.Version`

Reply:
All the test cases could change to `*version.Version`.

Reply:
updated.

Reply:
Thanks.

Then, the `*version.MustParseSemantic` could change to `version.MustParseSemantic` and the variable that used it does not need to deference `&`.

Reply:
updated.